### ISSUE TITLE
Render 'each' with 'fill()' when flattening models with -d=nonfScalarize

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#10795]
Consider the following MWE
```modelica
package TestEach
  model M1
    parameter Real p;
    Real x;
  equation
    der(x) = 1;
  end M1;
  
  model M2
    M1 m1[10](each p = 2,
              x(each start = 1,
                each fixed = true));
    M1 m11[10,10](each p = 2);
  end M2;
  
  model M3
    M2 m2[3];
  end M3;
end TestEach;
```
When flattening the models M2 and M3  with flags
```
-f -d=nonfScalarize,arrayConnect,combineSubscripts,printRecordTypes,evaluateAllParameters,vectorizeBindings --newBackend --showStructuralAnnotations
```
I get
```modelica
model 'TestEach.M2'
  Real[10] 'm1.x'(fixed = {true, true, true, true, true, true, true, true, true, true}, start = fill(1.0, 10));
  parameter Real[10] 'm1.p' = {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0};
  Real[10, 10] 'm11.x';
  parameter Real[10, 10] 'm11.p' = {{2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}};
equation
...
end 'TestEach.M2';
```
```modelica
model 'TestEach.M3'
  parameter Real[3, 10, 10] 'm2.m11.p' = {{{2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}}, {{2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}}, {{2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}}};
  Real[3, 10, 10] 'm2.m11.x';
  parameter Real[3, 10] 'm2.m1.p' = {{2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}, {2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0}};
  Real[3, 10] 'm2.m1.x'(fixed = {{true, true, true, true, true, true, true, true, true, true}, {true, true, true, true, true, true, true, true, true, true}, {true, true, true, true, true, true, true, true, true, true}}, start = fill(1.0, 3, 10));
equation
...
end 'TestEach.M3';
```
Now, according to the draft for Base Modelica ([MCP 0031](https://github.com/modelica/ModelicaSpecification/tree/MCP/0031/RationaleMCP/0031)), `each` should not be used. However, the current output is uselessly verbose, as it renders the result of `each` by large literal arrays that are by construction filled with the same value. 

I would propose to use `fill()` consistently in all those cases, as already done for `start` attributes, thus obtaining
```modelica
model 'TestEach.M2'
  Real[10] 'm1.x'(fixed = fill(true, 10), start = fill(1.0, 10));
  parameter Real[10] 'm1.p' = fill(2.0, 10);
  Real[10, 10] 'm11.x';
  parameter Real[10, 10] 'm11.p' = fill(2.0, 10, 10);
equation
...
end 'TestEach.M2';
```
and
```modelica
model 'TestEach.M3'
  parameter Real[3, 10, 10] 'm2.m11.p' = fill(2.0, 3, 10, 10);
  Real[3, 10, 10] 'm2.m11.x';
  parameter Real[3, 10] 'm2.m1.p' = fill(2.0, 3, 10);
  Real[3, 10] 'm2.m1.x'(fixed = fill(true, 3, 10), start = fill(1.0, 3, 10));
equation
...
end 'TestEach.M3';

**Comment by casella** [comment]
Adding @mscuttari to the loop.

**Comment by perost** [comment]
Possibly fixed in #10800, we'll see if it breaks any of the existing non-scalarized test cases. The fix isn't very good though, because I just turned off evaluation of `fill` calls with constant arguments when evaluating bindings during the evaluation phase.

Ideally we should never expand range expressions, `fill`, and other similar array constructions, and just have them transparently behave like arrays instead. But that's unfortunately hard to do in MetaModelica since it doesn't have any kind of interfaces.

**Comment by casella** [comment]
I think it's good enough for our current needs. We'll see. Thanks!
